src/values/basic_value_use.rs

```diff
@@ -4,15 +4,19 @@ use llvm_sys::prelude::LLVMUseRef;
 use std::marker::PhantomData;
 
 use crate::basic_block::BasicBlock;
-use crate::values::{AnyValueEnum, BasicValueEnum};
+use crate::values::{AnyValueEnum, BasicValueEnum, MetadataValue};
 
-/// Either [BasicValueEnum] or [BasicBlock].
+/// Either [BasicValueEnum], a [BasicBlock] or a [Metadata].
+///
+/// Note that [Metadata] variants are only constructed for LLVM 17+.
```

Comment:
You should use cfg/cfg_attr here to make it so that the documentation depends on the version. It shouldn't mention `metadata` in versions that it doesn't appear in.